File: web/src/api/dockerCluster.js
import service from '@/utils/request'

// @Tags DockerCluster
// @Summary 创建DockerCluster
export const createDockerCluster = (data) => {
  return service({
    url: '/dockerCluster/createDockerCluster',
    method: 'post',
    data
  })
}

// @Tags DockerCluster
// @Summary 删除DockerCluster
export const deleteDockerCluster = (data) => {
  return service({
    url: '/dockerCluster/deleteDockerCluster',
    method: 'delete',
    data
  })
}

// @Tags DockerCluster
// @Summary 批量删除DockerCluster
export const deleteDockerClusterByIds = (data) => {
  return service({
    url: '/dockerCluster/deleteDockerClusterByIds',
    method: 'delete',
    data
  })
}

// @Tags DockerCluster
// @Summary 更新DockerCluster
export const updateDockerCluster = (data) => {
  return service({
    url: '/dockerCluster/updateDockerCluster',
    method: 'put',
    data
  })
}

// @Tags DockerCluster
// @Summary 用id查询DockerCluster
export const findDockerCluster = (params) => {
  return service({
    url: '/dockerCluster/findDockerCluster',
    method: 'get',
    params
  })
}

// @Tags DockerCluster
// @Summary 分页获取DockerCluster列表
export const getDockerClusterList = (params) => {
  return service({
    url: '/dockerCluster/getDockerClusterList',
    method: 'get',
    params
  })
}

// @Tags DockerCluster
// @Summary 获取Docker集群凭证
export const getDockerClusterCredentials = (params) => {
  return service({
    url: '/dockerCluster/getDockerClusterCredentials',
    method: 'get',
    params
  })
}

// @Tags DockerCluster
// @Summary 获取所有Docker集群（用于下拉选择）
export const getAllDockerClusters = () => {
  return service({
    url: '/dockerCluster/getAllDockerClusters',
    method: 'get'
  })
}

File: server/router/dockerCluster/dockerCluster.go
package dockerCluster

import (
	"github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DockerClusterRouter struct {
}

// InitDockerClusterRouter 初始化 DockerCluster 路由信息
func (s *DockerClusterRouter) InitDockerClusterRouter(Router *gin.RouterGroup) {
	dockerClusterRouter := Router.Group("dockerCluster").Use(middleware.OperationRecord())
	dockerClusterRouterWithoutRecord := Router.Group("dockerCluster")
	{
		dockerClusterRouter.POST("createDockerCluster", v1.ApiGroupApp.DockerClusterApi.CreateDockerCluster)             // 新建DockerCluster
		dockerClusterRouter.DELETE("deleteDockerCluster", v1.ApiGroupApp.DockerClusterApi.DeleteDockerCluster)         // 删除DockerCluster
		dockerClusterRouter.DELETE("deleteDockerClusterByIds", v1.ApiGroupApp.DockerClusterApi.DeleteDockerClusterByIds) // 批量删除DockerCluster
		dockerClusterRouter.PUT("updateDockerCluster", v1.ApiGroupApp.DockerClusterApi.UpdateDockerCluster)              // 更新DockerCluster
	}
	{
		dockerClusterRouterWithoutRecord.GET("findDockerCluster", v1.ApiGroupApp.DockerClusterApi.FindDockerCluster)                // 根据ID获取DockerCluster
		dockerClusterRouterWithoutRecord.GET("getDockerClusterList", v1.ApiGroupApp.DockerClusterApi.GetDockerClusterList)          // 获取DockerCluster列表
		dockerClusterRouterWithoutRecord.GET("getDockerClusterCredentials", v1.ApiGroupApp.DockerClusterApi.GetDockerClusterCredentials) // 获取集群凭证
		dockerClusterRouterWithoutRecord.GET("getAllDockerClusters", v1.ApiGroupApp.DockerClusterApi.GetAllDockerClusters)          // 获取所有集群（下拉选择）
	}
}
